### ISSUE TITLE
EEPROM.commit() blocks interrupts for 30ms

### ISSUE THREAD

**Issue by kachurovskiy** [espressif/arduino-esp32#8169]
### Board

ESP32-S3-WROOM-1U-N16R2

### Device Description

See https://github.com/kachurovskiy/nanoels/tree/main/h4 but I don't think it's relevant.

### Hardware Configuration

My board schematic is available on https://github.com/kachurovskiy/nanoels/tree/main/h4#main-board-schematic but I don't think it's relevant.

### Version

v2.0.9

### IDE Name

Arduino IDE

### Operating System

Windows 10

### Flash frequency

Espressif Systems 

### PSRAM enabled

no

### Upload speed

115200

### Description

With `commit()` call it looks like this:

![image](https://user-images.githubusercontent.com/517919/236675661-fe32e51b-ee30-43f1-b07f-579a6d6776a5.png)

Without `commit()` it's all working well:

![image](https://user-images.githubusercontent.com/517919/236675678-33c75575-3b1c-42ba-9566-2ca19c1ac702.png)

This problem also affects hardware PCNT pulse counter and likely other interrupt logic.

### Sketch

```cpp
#define ENC_A 15
#define OUT 11

#include <EEPROM.h>

void IRAM_ATTR spinEnc() {
  digitalWrite(OUT, digitalRead(ENC_A));
}

void setup() {
  EEPROM.begin(256);
  pinMode(ENC_A, INPUT);
  pinMode(OUT, OUTPUT);
  attachInterrupt(digitalPinToInterrupt(ENC_A), spinEnc, CHANGE);
}

void loop() {
  EEPROM.write(0, digitalRead(ENC_A));
  // EEPROM.commit(); // uncommend this line to get 30ms periods of ignored interrupts
  delay(10);
}
```

### Debug Message

```plain
Not relevant.
```

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by mrengineer7777** [comment]
Consider the following:
1) `EEPROM.write()` copies a single byte into an array to be written.
2) `EEPROM.commit()` writes buffered data to flash.
3) Writing to flash takes time.
4) Processor is executing code from flash.
5) Processor cannot access flash during a flash write (depends on processor architecture).

What do you expect to happen?

For a further analysis you would have to dig into the NVS libraries in the IDF.

Workarounds:
1) Schedule your commit for a time when interrupts can be blocked.
2) Save to an external EEPROM chip.
3) Save to an external battery backed RAM chip.

Note: the same issue likely occurs when writing files to SPIFFS.

**Comment by kachurovskiy** [comment]
I agree with your analysis - the goal of this bug is to ensure this behavior is well documented in EEPROM, Preferences and interrupts documentation - it might not be sufficiently emphasized at the moment.

**Comment by lbernstone** [comment]
https://github.com/espressif/arduino-esp32/tree/master/libraries/EEPROM

**Comment by Parsaabasi** [comment]
Hello, 

Due to the overwhelming volume of issues currently being addressed, we have decided to close the previously received tickets. If you still require assistance or if the issue persists, please don't hesitate to reopen the ticket.

Thanks.
